onmt/modules/GlobalAttention.py

```diff
@@ -39,18 +39,17 @@ def __init__(self, dim, coverage=False, attn_type="dotprod"):
 
         self.dim = dim
         self.attn_type = attn_type
-        assert (self.attn_type in ["dotprod", "mlp"]), (
+        assert (self.attn_type in ["dotprod", "general", "mlp"]), (
                 "Please select a valid attention type.")
 
-        if self.attn_type == "dotprod":
+        if self.attn_type == "general":
```

Comment:
I think these names should be more clear. Maybe there can be a comment with the math for each of them? Alternatively we can sync with OpenNMT Lua